### PR TITLE
MODREQMED-40 User, Proxy, and Item data are stored and returned if reference records are deleted

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose

Resolves: [MODREQMED-40](https://folio-org.atlassian.net/browse/MODREQMED-40)

### CODE REVIEW

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java [hunk 1]
```diff
@@ -119,57 +125,112 @@ private MediatedRequestContext buildRequestContext(MediatedRequest request) {
...
-        .proxyGroup(proxyGroup);
-    }
+  private void fetchInventoryInstance(SearchInstance searchInstance,
+    MediatedRequestContextBuilder ctxBuilder, MediatedRequest request) {
+
+    log.info("fetchInventoryInstance: searchInstance: {}", searchInstance);
+    executionService.executeAsyncSystemUserScoped(searchInstance.getTenantId(),
+      () -> ctxBuilder.instance(inventoryService.fetchInstance(searchInstance.getId())));
 
     if (request.getItemId() != null) {
```
Comment (imerabishvili) [inline src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java:170]: Similar thing:
```
if (request.getItemId() == null) {
  return;
}
...
```

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java [hunk 2]
```diff
@@ -119,57 +125,112 @@ private MediatedRequestContext buildRequestContext(MediatedRequest request) {
...
+      } else {
+        log.info("fetchInventoryItem:: inventoryItem {} not found", searchItem.getId());
+        ctxBuilder.item(new Item().barcode(request.getItem().getBarcode()));
+      }
+    });
+  }
+
+  private void fetchProxyUser(MediatedRequest request, MediatedRequestContextBuilder ctxBuilder) {
+
+    if (request.getProxyUserId() != null) {
```
Comment (imerabishvili) [inline src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java:202]: Same: please invert if condition

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java [hunk 3]
```diff
@@ -119,57 +125,112 @@ private MediatedRequestContext buildRequestContext(MediatedRequest request) {
...
+            .lastName(medReqProxy.getLastName())
+            .middleName(medReqProxy.getMiddleName())));
+      }
     }
+  }
+
+  private void fetchPickupServicePoint(MediatedRequest request,
+    MediatedRequestContextBuilder ctxBuilder) {
 
     if (request.getPickupServicePointId() != null) {
```
Comment (imerabishvili) [inline src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java:222]: Please invert if condition

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java [hunk 4]
```diff
@@ -197,12 +258,14 @@ private static void extendRequester(MediatedRequestContext context) {
   private static void addRequesterGroup(MediatedRequestContext context) {
     log.info("addRequesterGroup:: adding requester user group data");
     UserGroup userGroup = context.requesterGroup();
-    context.request()
-      .getRequester()
-      .patronGroup(new MediatedRequestRequesterPatronGroup()
-        .id(userGroup.getId())
-        .group(userGroup.getGroup())
-        .desc(userGroup.getDesc()));
+    if (userGroup != null) {
```
Comment (imerabishvili) [inline src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java:261]: Please invert if condition

**Review by alexanderkurash — APPROVED** [review]

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java [hunk 5]
```diff
@@ -119,57 +125,112 @@ private MediatedRequestContext buildRequestContext(MediatedRequest request) {
...
+      .orElseGet(() -> createFallbackUser(request.getRequester()));
+  }
+
+  private void handleSearchInstances(List<SearchInstance> searchInstances,
+    MediatedRequestDetailsServiceImpl.MediatedRequestContext.MediatedRequestContextBuilder ctxBuilder,
+    MediatedRequest request) {
 
-    var searchInstances = searchClient.searchInstance(request.getInstanceId())
-      .getInstances();
     if (searchInstances == null || searchInstances.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java:149]: done

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java (hunk 1 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java]: done

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java (hunk 2 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java]: done

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java (hunk 3 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java]: done

File: src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java (hunk 4 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/mr/service/impl/MediatedRequestDetailsServiceImpl.java]: done

**Pull request merged**